test/Hamiltonians.jl

```diff
@@ -1669,3 +1668,26 @@ end
     ]
     [@test dimension(addr) == dimension(typeof(addr)) for addr in addresses]
 end
+
+@testset "ExtendedHubbardReal1D boundary conditions" begin
+    for H in (
+        ExtendedHubbardReal1D(FermiFS((1,0,1,0)), v=6, t=2.0, boundary_condition=:twisted),
+        ExtendedHubbardReal1D(FermiFS((1,0,1,0)), v=6, t=2.0, boundary_condition=:hard_wall),
+        ExtendedHubbardReal1D(FermiFS((1,0,1,0,1,0,1,0,1,0,1)), v=6, t=2.0, boundary_condition=:twisted),
+        ExtendedHubbardReal1D(FermiFS((1,0,1,0,1,0,1,0,1,0,1)), v=6, t=2.0, boundary_condition=:hard_wall))
+        
+        addr = starting_address(H)
+        H1 = ExtendedHubbardReal1D(addr, v=6, t=2.0)
+        addr2, me = get_offdiagonal(H1, addr, 2)
+        if H.boundary_condition == :twisted
+            @test get_offdiagonal(H, addr, 2)[2] == - me
+            @test diagonal_element(H, addr) == diagonal_element(H1, addr)
+        elseif H.boundary_condition == :hard_wall
+            @test get_offdiagonal(H, addr, 2)[2] == 0.0
+        end
+        @test get_offdiagonal(H, addr, 2)[1] == addr2
+    end
+    H1 = ExtendedHubbardReal1D(BoseFS(1,1,1,1))
+    H2 = ExtendedHubbardReal1D(BoseFS(1,1,1,1); boundary_condition=:hrad_wall)
+    @test sparse(H1) == sparse(H2)
```

Comment:
Ha ha, this is not supposed to work 🤣 

What @mtsch meant with his comment was to make the code check whether the arguments are sensible and throw an error if they don't. Then you can test the behaviour like this:
```suggestion
    @test_thows ArgumentError ExtendedHubbardReal1D(BoseFS(1,1,1,1); boundary_condition=:hrad_wall)
```